src/modules/userblacklist/userblacklist.c

```diff
@@ -474,6 +517,38 @@ static int check_globalblacklist_fixup(void** param, int param_no)
 	return 0;
 }
 
+static int ki_check_globalblacklist(sip_msg_t *msg)
+{
+	char * table = globalblacklist_table.s;
+	struct check_blacklist_fs_t* arg = NULL;
+
+	if (!table) {
+		LM_ERR("no table name\n");
+		return -1;
+	}
+	/* try to add the table */
+	if (add_source(table) != 0) {
+		LM_ERR("could not add table");
+		return -1;
+	}
+
+	gnode = table2dt(table);
+	if (!gnode) {
+		LM_ERR("invalid table '%s'\n", table);
+		return -1;
+	}
+
+	arg = pkg_malloc(sizeof(struct check_blacklist_fs_t));
```

Comment:
The `arg` is allocated in pkg, shouldn't it be freed after the call of `check_blackist()`? There are similar cases in the other functions. I am not familiar with the code of the module, but upon quick check there are at least some error cases inside `check_blackist()` that just return back without any pkg free.

Reply:
Hi Daniel, i've just added the missing pkg_free before the functions returns. Sorry of the mistake but it was not my code.

Reply:
Maybe the pkg_free() has to be done in the functions you added (the kemi functions), otherwise the last change you did seems to affect the old code which was not reported to have issues. It cna be that now it frees for the kamailio.cfg native functions, when it should not happen.

Reply:
I've done the change as requested. Please check the commit and let me know.